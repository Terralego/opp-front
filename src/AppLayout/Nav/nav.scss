@import '../../variables.scss';

.navBar {
  display: flex;
  flex-direction: column;
  height: auto;
  flex: 1;
  padding: 10px 0;

  // Override Blueprintjs
  .bp3-navbar {
    &-heading {
      .logo {
        width: 80px;
      }
    }
  }

  .header-button-label {
    display: none;
  }

  &__group {
    margin-top: 20px;
    background: rgba(255, 255, 255, 0.1);
    box-shadow: inset 2px 1px 1px 0 rgba(0, 0, 0, 0.5);
    flex-direction: column;
    height: auto;

    ul {
      list-style: none;
      text-align: center;
      width: 100%;
      margin: 0;
      padding: 0;
    }
    li {
      width: 100%;
      position: relative;
    }

    .header-link {
      width: 100%;
      margin: 10px;

      button {
        display: block;
        position: relative;
        margin: 0 auto;
        padding: 0.5em 1em;
        z-index: 1;
      }

      &.active {
        &::before {
          content: '';
          display: block;
          position: absolute;
          top: 0;
          right: 0;
          bottom: 0;
          left: 0;
          background-color: $solidgray;
        }
        &::after {
          content: '';
          position: absolute;
          right: 0;
          top: calc(50% - 5px);
          background-image: url('/images/nav__arrow-right.svg');
          background-repeat: no-repeat;
          display: inline-block;
          width: 15px;
          height: 20px;
          vertical-align: middle;
        }
      }
    }
  }
}
