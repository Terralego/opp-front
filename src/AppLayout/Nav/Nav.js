import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { NavLink } from 'react-router-dom';

import { MainMenu } from '@terralego/core/components/MainMenu';

import Logo from '../../components/Logo';
import './nav.scss';

const getLinkProps = link =>
  !link.startsWith('http') && {
    link: {
      component: NavLink,
      linkProps: {
        exact: true,
        hrefAttribute: 'to',
      },
    },
  };

export const MainNav = ({ logo, logoUrl, resetMapInitialState, toggleSearchFoldedState }) => {
  const { t } = useTranslation();

  const menu = useMemo(
    () => ({
      navHeader: {
        href: logoUrl,
        icon: <Logo src={logo} alt={t('menu.logoName')} />,
        label: t('menu.logoLink'),
        onClick: resetMapInitialState,
        ...getLinkProps(logoUrl),
      },
      navItems: [
        [
          {
            href: '/',
            icon: 'search',
            id: 'search',
            onClick: toggleSearchFoldedState,
            label: t('menu.search'),
            ...getLinkProps('/'),
          },
        ],
      ],
    }),
    [logo, logoUrl, resetMapInitialState, t, toggleSearchFoldedState],
  );

  return <MainMenu className="main__header" {...menu} />;
};

MainNav.propTypes = {
  logo: PropTypes.string,
  logoUrl: PropTypes.string,
  resetMapInitialState: PropTypes.func,
  toggleSearchFoldedState: PropTypes.func,
};

MainNav.defaultProps = {
  logo: undefined,
  logoUrl: '/',
  resetMapInitialState: () => {},
  toggleSearchFoldedState: () => {},
};

export default MainNav;
