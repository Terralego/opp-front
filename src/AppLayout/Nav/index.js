import { connectSettingsProvider } from '../../components/SettingsProvider';
import { connectAppProvider } from '../context';

import Nav from './Nav';

export default connectSettingsProvider(({ env: { theme: { logo, logoUrl } } }) => ({
  logo,
  logoUrl,
}))(
  connectAppProvider({
    toggleSearchFoldedState: 'actions.toggleSearchFoldedState',
    resetMapInitialState: 'actions.resetMapInitialState',
  })(Nav),
);
