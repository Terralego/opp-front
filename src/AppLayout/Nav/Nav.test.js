import React from 'react';
import renderer from 'react-test-renderer';

import Nav from './Nav';

jest.mock('react-router-dom', () => ({
  NavLink: ({ children, to }) => <a href={to}>{children}</a>,
}));

it('should render', () => {
  const tree = renderer.create(<Nav />).toJSON();
  expect(tree).toMatchSnapshot();
});
