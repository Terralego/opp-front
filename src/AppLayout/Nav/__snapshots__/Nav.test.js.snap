// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="bp3-navbar navBar bp3-dark main__header"
>
  <div
    className="bp3-navbar-heading navBar__header"
  >
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/"
        >
          <button
            className="bp3-button bp3-minimal header-button"
            id={null}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <span>
              menu.logoName
            </span>
          </button>
        </a>
      </span>
    </span>
  </div>
  <div
    className="bp3-navbar-group bp3-align-left navBar__group"
  >
    <ul>
      <li>
        <span
          className="bp3-popover-wrapper"
        >
          <span
            className="bp3-popover-target"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <a
              href="/"
            >
              <button
                className="bp3-button bp3-minimal header-button"
                id="search"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                type="button"
              >
                <span
                  className="bp3-icon bp3-icon-search"
                  icon="search"
                >
                  <svg
                    data-icon="search"
                    height={16}
                    viewBox="0 0 16 16"
                    width={16}
                  >
                    <desc>
                      search
                    </desc>
                    <path
                      d="M15.55 13.43l-2.67-2.68a6.94 6.94 0 001.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 102.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
                      fillRule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </a>
          </span>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
